pkg/providers/instancetype/offering/provider.go

```diff
@@ -218,10 +235,39 @@ func (p *DefaultProvider) cacheKeyFromInstanceType(it *cloudprovider.InstanceTyp
 		&hashstructure.HashOptions{SlicesAsSets: true},
 	)
 	return fmt.Sprintf(
-		"%s-%016x-%016x-%d",
+		"%s-%016x-%016x",
 		it.Name,
 		zonesHash,
 		capacityTypesHash,
-		p.unavailableOfferings.SeqNum,
 	)
 }
+
+func (p *DefaultProvider) updateOfferingAvailability() {
+	for k, v := range p.cache.Items() {
+		var updatedOfferings []*cloudprovider.Offering
+		// Extract instance type name from cache key
+		itName := strings.Split(k, "-")[0]
```

Comment:
Is there a better way to do this? This is a bit fragile, but perhaps there is as good as it gets with stringly-typed cache keys

Reply:
You should be able to derive from the instance type resolved from the cache, no? From `v.Object`